### ISSUE TITLE
Reliable fetch inefficient with many queues

### ISSUE THREAD

**Issue by mperham** [mperham/sidekiq#1769]
Reliable fetch polls each queue before blocking.  With 100 queues, you'll see 100 redis operations before blocking on the first for one second.  Repeat.

Switch to Lua so we don't have the network round trip issue.  It looks like we won't get a command to solve this use case, see antirez/redis#1785.

**Comment by mperham** [comment]
Gonna close this as it's impossible to implement without support from Redis, which does not seem to be coming anytime soon.

**Comment by Kache** [comment]
Is this issue is relevant to super_fetch, as mentioned in [wiki/Advanced-Options](https://github.com/mperham/sidekiq/wiki/Advanced-Options)? If so, I was wondering if `RANDOMKEY` could help alleviate this and other antirez/redis#1785 related issues.

Store all queues as lists within an independent Redis db without any other keys, and have processes poll using `RANDOMKEY`, followed by a `RPOPLPUSH` attempt. There can still be contention, ofc, but would that be better than loop-polling every queue for a high number of queues?

In other words, would it be the case that (and is it a correct comparison):

```
RANDOMKEY + RPOPLPUSH: O(1) * # of Producers + O(1)
loopQueues(RPOPLPUSH): O(1) * # of Producers * # of Queues
```

Definitely not very elegant, since it'd require Sidekiq to have a second Redis db, and weighted random sampling would be lost.
